service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/OrdersNeededAboutToSubmitCallbackController.java

```diff
@@ -55,6 +71,23 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(
             data.put("otherOrderType", "NO");
         }
 
+        Court selectedCourt = getCourtSelection(courtID);
+
+        if (Objects.nonNull(selectedCourt)) {
+            data.remove("court");
+            data.put("court", selectedCourt);
+        }
+
+        if (Objects.equals(ordersText, "ordersSolicitor")) {
+            data.remove("orders");
+            data.put("orders", data.get("ordersSolicitor"));
+            data.remove("ordersSolicitor");
```

Comment:
May need to double check this one but this may not clear the field, night need to set to null instead